assets/js/googlesitekit/datastore/user/key-metrics.js

```diff
@@ -367,16 +366,18 @@ const baseSelectors = {
 			const keyMetricSettings =
 				select( CORE_USER ).getKeyMetricsSettings();
 
-			const isGA4Connected =
-				select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
-
-			const detectedEvents = isGA4Connected
-				? select( MODULES_ANALYTICS_4 ).getDetectedEvents()
-				: [];
-
-			const showConversionTailoredMetrics =
-				keyMetricSettings?.includeConversionTailoredMetrics ||
-				includeConversionTailoredMetrics;
+			const showConversionTailoredMetrics = ( events ) => {
```

Comment:
Since tailored metrics are "phantom" like, not saved, and just appearing/disappearing based on conditions set, once events are lost, they will disappear from dashboard, and we want to show lost events banner only, not magically remove key metrics. For that purpose I refactored and simplified this logic a bit, to determine each ACR metric based on the detected event that it is relating to, and allow for override - which is used in preview in confirm modal of purpose change, so we can properly show ACR metrics included granularly depending if user has those events detected. Also I changed  `includeConversionTailoredMetrics` setting type to accept array of events - kinda like a snapshot to persist the KMW when events are lost, so they remain on the dashboard and only show notification that events are lost